### PR TITLE
fix #1794 unexcepted exception can roll back

### LINKED ISSUE #1794: rollbackFor does not work properly
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅲ. Describe what you expected to happen
方法设置：    @GlobalTransactional(name = "fsp-create-order", rollbackFor = RuntimeException.class)

异常抛出：Exception

结果：应该不回滚，但是回滚了，我觉得rollbackOn方法有问题

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅴ. Anything else we need to know?
public boolean rollbackOn(Throwable ex) {

        RollbackRule winner = null;
        int deepest = Integer.MAX_VALUE;

        if (this.rollbackRules != null) {
            for (RollbackRule rule : this.rollbackRules) {
                int depth = rule.getDepth(ex);
                if (depth >= 0 && depth < deepest) {
                    deepest = depth;
                    winner = rule;
                }
            }
        }

        return winner == null || !(winner instanceof NoRollbackRule);
    }

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by github-ygy**
### Ⅰ. Describe what this PR did
fix #1794 unexcepted exception can roll back

### Ⅳ. Describe how to verify it
TransactionInfoTest#testRollBackOn

### CODE REVIEW

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ public boolean rollbackOn(Throwable ex) {
         int deepest = Integer.MAX_VALUE;
 
         if (this.rollbackRules != null) {
+            winner = NoRollbackRule.DEFAUL_NO_ROLLBACK_RULE;
```
Comment (jsbxyyx) [inline tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: the rollbackRules is not null but it is empty, not do roll back?

File: tm/src/main/java/io/seata/tm/api/transaction/NoRollbackRule.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
  */
 public class NoRollbackRule extends RollbackRule {
 
+    public static final NoRollbackRule DEFAUL_NO_ROLLBACK_RULE = new NoRollbackRule(Throwable.class);
```
Comment (xingfudeshi) [inline tm/src/main/java/io/seata/tm/api/transaction/NoRollbackRule.java]: DEFAUL->DEFAULT

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 1 above)
Comment (xingfudeshi) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: > the rollbackRules is not null but it is empty, not do roll back?

When might it be empty?

Comment (jsbxyyx) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: image

Comment (xingfudeshi) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: OK.Then this.rollbackRules != null should be replaced with CollectionUtils.isNotEmpty.

File: tm/src/main/java/io/seata/tm/api/transaction/NoRollbackRule.java (hunk 2 above)
Comment (github-ygy) [reply tm/src/main/java/io/seata/tm/api/transaction/NoRollbackRule.java]: done

File: tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java (hunk 1 above)
Comment (github-ygy) [reply tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java]: done ,thanks for @jsbxyyx

**Pull request merged**